qiskit_experiments/library/randomized_benchmarking/clifford_utils.py

```diff
@@ -375,328 +294,322 @@ def _unpack_num(num, sig):
-
-    def _transpile_cliff_layer_2(self):
-        """Layer 2 consists of a Pauli gate on each qubit {Id, X, Y, Z}.
-        Number of Cliffords == 16."""
-        if self._transpiled_cliff_layer[2] != []:
-            return
-
-        pauli = ("i", XGate(), YGate(), ZGate())
-        for p0, p1 in itertools.product(pauli, pauli):
-            qc = QuantumCircuit(2)
-            if p0 != "i":
-                qc.append(p0, [0])
-            if p1 != "i":
-                qc.append(p1, [1])
-
-            transpiled = transpile(
-                qc, optimization_level=1, basis_gates=self.basis_gates, backend=self._backend
-            )
-            self._transpiled_cliff_layer[2].append(transpiled)
+        _append_v_w(qc, v0, v1)
+        circuits.append(qc)
 
-    def create_random_clifford(self, rng: Generator) -> QuantumCircuit:
-        """For 1-qubit, select a random transpiled Clifford.
-
-         Returns:
-            A random Clifford represented as a :class:`QuantumCircuit`.
+    for v0, v1 in itertools.product(v_w_gates, v_w_gates):
+        qc = QuantumCircuit(2)
+        qc.cx(0, 1)
+        qc.cx(1, 0)
+        _append_v_w(qc, v0, v1)
+        circuits.append(qc)
+
+    qc = QuantumCircuit(2)  # swap at the end
+    qc.cx(0, 1)
+    qc.cx(1, 0)
+    qc.cx(0, 1)
+    circuits.append(qc)
+    return circuits
+
+
+def _create_cliff_2q_layer_2():
+    """Layer 2 consists of a Pauli gate on each qubit {Id, X, Y, Z}.
+    Number of Cliffords == 16."""
+    circuits = []
+    pauli = ("i", XGate(), YGate(), ZGate())
+    for p0, p1 in itertools.product(pauli, pauli):
+        qc = QuantumCircuit(2)
+        if p0 != "i":
+            qc.append(p0, [0])
+        if p1 != "i":
+            qc.append(p1, [1])
+        circuits.append(qc)
+    return circuits
+
+
+_CLIFFORD_LAYER = (
+    _create_cliff_2q_layer_0(),
+    _create_cliff_2q_layer_1(),
+    _create_cliff_2q_layer_2(),
+)
+_NUM_LAYER_0 = 36
+_NUM_LAYER_1 = 20
+_NUM_LAYER_2 = 16
 
-        Raises:
-            QiskitError: if the number of qubits is greater than 2.
 
-        """
-        if rng is None:
-            rng = default_rng()
-        if isinstance(rng, int):
-            rng = default_rng(rng)
-        if self.num_qubits == 1:
-            rand = rng.integers(self.NUM_CLIFFORD_1_QUBIT)
-            return self._transpiled_cliffords_1q[rand]
-        elif self.num_qubits == 2:
-            rand = rng.integers(self.NUM_CLIFFORD_2_QUBIT)
-            return self.create_cliff_from_num(num=rand)
-        else:
-            raise QiskitError("create_random_clifford is not supported for more than 2 qubits")
+@lru_cache(maxsize=None)
+def _transformed_clifford_layer(
+    layer: int, index: Integral, basis_gates: Tuple[str, ...]
+) -> QuantumCircuit:
+    # Return the index-th quantum circuit of the layer translated with the basis_gates.
+    # The result is cached for speed.
+    return _transform_clifford_circuit(_CLIFFORD_LAYER[layer][index], basis_gates)
```

Comment:
No idea. This function may return different circuit object from input circuit object, so not sounds very bad to me. But I have no strong opinion or preference here. If you have any preference, I'll change to it. If not, I'll leave it as is. This is private so it would be easy to change it when we think of a better name. Any preference? @nkanazawa1989